bin/migrate_vl_to_ldl.rb

```diff
@@ -0,0 +1,69 @@
+CSV_FILE_NAME = "vl_to_ldl_migration_#{Time.now.strftime('%Y%m%d')}.csv"
+HIV_PROGRAM = Program.where(name: "HIV PROGRAM").first
+
+# all test results with 1 as value numeric and = as value_modifier for HIV program
+def test_results
+  Observation.joins(:encounter).where(encounter: {program: HIV_PROGRAM }, obs: {value_modifier: '=', value_numeric: 1})
+end
+
+def superuser
+  User.find(1)
+end
+
+def start
+  records = test_results
+  @records_count = records.count
+  return puts "No records to migrate" if @records_count == 0
+  setup
+  puts "Starting migration of #{test_results.count} test results"
+  records.each do |obs|
+    new_record = create_dup obs
+    void_record obs
+    log_record new_record, obs
+    progress
+  end
+  cleanup
+  puts "\nMigration complete"
+end
+
+def progress
+  @count += 1
+  STDOUT.write "\r#{@count*100/@records_count}% done"
+end
+
+def create_dup(obs)
+  new_obs = obs.dup
+  new_obs.value_numeric = nil
+  new_obs.value_text = 'LDL'
+  new_obs.value_modifier = '='
+  new_obs.uuid = nil
+  new_obs.creator = superuser.id
+  Observation.create new_obs.attributes
+end
+
+def void_record(obs)
```

Comment:
We have the void method on all voidable methods. Something like this should work
```bash
obs.void('your reason')
```